src/main/java/com/nutomic/syncthingandroid/fragments/SettingsFragment.java

```diff
@@ -236,11 +249,19 @@ public boolean onPreferenceChange(Preference preference, Object o) {
                     : R.string.always_run_in_background_disabled);
             mSyncOnlyCharging.setEnabled(value);
             mSyncOnlyWifi.setEnabled(value);
+            mSyncOnlyOnSSIDs.setEnabled(mSyncOnlyWifi.isEnabled() && mSyncOnlyWifi.isChecked());
```

Comment:
You can remove the isEnabled() check, because we uncheck the preference when we disable it.